### DISCUSSION TITLE
Output units for synthetic seismograms

### DISCUSSION THREAD

**Discussion by ammcpherson** [SPECFEM/specfem3d#1567] [SPECFEM3D_GLOBE General]
Hello everybody,

I was wondering if anyone could tell me what the output units are for the synthetic seismograms generated in SPECFEM3D_Globe. I see that the SAC headers have been populated with information, and it appears that the unit (for a displacement synthetic) is "nm' - nanometer?

Thanks,
Amanda

**Comment by danielpeter** [comment]
hi Amanda,

the globe version always outputs displacement seismograms with units in [m]. in SAC files, you can indicate what these trace values should be by the variable IDEP. options given by SAC (https://ds.iris.edu/files/sac-manual/manual/file_format.html) as type of dependent variable are:
- IUNKN (Unknown)
- IDISP (Displacement in nm)
- IVEL (Velocity in nm/sec)
- IACC (Acceleration in nm/sec/sec)
- *NOTE* Others use SI units for IDISP, IVEL, IACC.
- IVOLTS (Velocity in volts)

SPECFEM3D_GLOBE chooses the setting `IDISP` as type in the output SAC files, but using SI units rather than nm.

**Reply by changcheng666li** [reply]
hi professor,

Can you tell me the output units of SAVE_SEISMOGRAMS_VELOCITY is nm/s or m/s?

Thanks,
Changcheng

**Reply by danielpeter** [reply]
you refer to the SPECFEM3D_Cartesian version. that one has no SAC output. 

but to answer your question what the Cartesian version would output: that's a bit more tricky. the input mesh in Cartesian could be given in any dimension. for example, in ultrasonics the mesh node positions could refer to micro/millimeters, or velocities given could be in km/s or other mixtures of dimension and seismic model units. as a user, you should be aware and consistent in what units you use for your inputs. the Cartesian version is in principle unaware of what unit then this refers to, and just provides the output for that dimension. 

we would advise to use SI units for mesh dimensions and seismic model values, because in that case the Cartesian version also outputs SI units where velocity would be [m/s].

**Reply by changcheng666li** [reply]
Thanks for your answer. I use the specfem3d_globe version. Is the units of velocity of this version also [m/s]?

**Reply by danielpeter** [reply]
SPECFEM3D_GLOBE only outputs displacement traces.

**Reply by changcheng666li** [reply]
Dear professor, Thank you for your reply.

**Reply by djavediqbal** [reply]
Can you please tell about units of output seismograms of SPECFEM2D

**Reply by bch0w** [reply]
@djavediqbal you can choose the unit of your output seismogram with the parameter `seismotype` 

https://github.com/SPECFEM/specfem2d/blob/98741db1a0c8082ca57364f5b17ea95df6cbf1c2/DATA/Par_file#L143-L145

**Comment by ammcpherson** [comment]
@danielpeter Thank you for clarifying!

**Comment by changcheng666li** [comment]
Dear professor, if I use the point force source as the source to calculate the seismic record, Should the unit of force in the input file be Newton or dyne?

**Comment by homnath** [comment]
For FORCESOLUTION, the unit of the force should be in SI unit, i.e., Newton. Make sure you use other units consistently.
For CMTSOLUTION, however, the units of moment tensor components are in dyne-cm.
________________________________

**Comment by changcheng666li** [comment]
Thank You!
